pkg/provisioner/ironic/ironic.go

```diff
@@ -414,6 +425,33 @@ func (p *ironicProvisioner) Provision(userData string) (result provisioner.Resul
 
 	result.RequeueAfter = provisionRequeueDelay
 
+	// FIXME(dhellmann): The Stein version of Ironic supports passing
+	// a URL. When we upgrade, we can stop doing this work ourself.
+	checksum := p.host.Spec.Image.Checksum
+	isURL, err := checksumIsURL(checksum)
+	if err != nil {
+		return result, errors.Wrap(err, "Could not understand image checksum")
+	}
+	if isURL {
+		p.log.Info("looking for checksum for image", "URL", checksum)
+		resp, err := http.Get(checksum)
+		if err != nil {
+			return result, errors.Wrap(err, "Could not fetch image checksum")
+		}
+		if resp.Close {
```

Comment:
I don't think this is related to closing the body of the message. It's related to closing the connection itself, which is different.

The docs for `Get` say "When err is nil, resp always contains a non-nil resp.Body. Caller should close resp.Body when done reading from it." So I think if the error is nil, you can just always close the body when you're done. No need to check this.

Reply:
Done